File: app/src/main/java/com/martynaskairys/walltip/SquareImageView.java
package com.martynaskairys.walltip;

import android.content.Context;
import android.util.AttributeSet;
import android.widget.ImageView;

/**
 * Created by Martynas on 2016-04-28.
 */
public class SquareImageView extends ImageView {

	public SquareImageView(Context context) {
		super(context);
	}

	public SquareImageView(final Context context, final AttributeSet attrs) {
		super(context, attrs);
	}

	public SquareImageView(final Context context, final AttributeSet attrs, final int defStyleAttr) {
		super(context, attrs, defStyleAttr);
	}

	@Override
	protected void onMeasure(int widthMeasureSpec, int heightMeasureSpec) {
		super.onMeasure(widthMeasureSpec, heightMeasureSpec);

		int width = getMeasuredWidth();
		setMeasuredDimension(width, width);
	}


}

File: app/src/main/java/com/martynaskairys/walltip/SquareImageView.kt
package com.martynaskairys.walltip

import android.content.Context
import android.util.AttributeSet
import android.widget.ImageView

/**
 * Created by Martynas on 2016-04-28.
 */
//TODO this class was used for thumbnail_activity thumb_pics to show square (if val width = measure width...)
//should I just use image_adapter_item or this class for the right measure

class SquareImageView : ImageView {

    constructor(context: Context) : super(context) {
    }

    constructor(context: Context, attrs: AttributeSet) : super(context, attrs) {
    }

    constructor(context: Context, attrs: AttributeSet, defStyleAttr: Int) : super(context, attrs, defStyleAttr) {
    }

    override fun onMeasure(widthMeasureSpec: Int, heightMeasureSpec: Int) {
        super.onMeasure(widthMeasureSpec, heightMeasureSpec)

        val height = measuredHeight
        setMeasuredDimension(height, height)
    }


}
